### DISCUSSION TITLE
How to use TailwindCSS together with MaterialUI  together in a project.

### DISCUSSION THREAD

**Discussion by jake0011** [tailwindlabs/tailwindcss#11464] [Help]
Hello guys, i am currently building an ERP system with basic functions for my final year project within a three months timeframe and i have just a month left, as such, i have resorted to patching up code for the UI instead of building something from scratch.

I have basic knowledge of HTML, CSS and JS but this is my first ever web project and my team has agreed on us using Next.js and TailwindCSS for the frontend and it has been so good so far. However, recently when i started copying and pasting code from different repos and sources, i came across this error upon successful build.

Error: My previously code written in TailwindCSS only, does not display the desired style but rather plain html and the new code i copied with MaterialUI styling happens to display fine.

I will share a page image for further clarity on what i am talking about.

The problem to me seems to be from the setup of materialui  as i did not configure it with tailwindcss from scratch of the project.

Kindly provide me with assistance on how to resolve this and also advice on how to write cleaner code and some best practices for this style of development( where alot of copying and pastings is done lol.)

![image](https://github.com/tailwindlabs/tailwindcss/assets/61389729/eff4a855-1cd1-47b8-89e3-362d3aad453c)

**Comment by samsamurai301** [comment]
Material UI has its own components which actually overrides the CSS. I prefer not to use both together. Once, I tried to do that but in the end, that was a useless thing. CSS in custom ways. There are plenty of UI suitable for that such as Flowbite & Headless UI or maybe Chakra and Ant Design you can search for that. But I will prefer you not to use both together because then there is no meaning for usefulness.

**Comment by tannerhallman** [comment]
Although I don't recommend it, If you have a use case for it (like gradual migration from MUI to tailwind), you can achieve it by using the [prefix](https://tailwindcss.com/docs/configuration#prefix) and the [important](https://tailwindcss.com/docs/configuration#important) config flags. This helps with css namespace conflicts that happen when you have a component library like that.

**Comment by siriwatknp** [comment]
The integration between Material UI (v6) and Tailwind CSS is a lot better. Please take a look at https://mui-treasury.com/?path=/docs/tailwind-preset--docs to get the preset.

**Reply by Badial5** [reply]
Really
